src/Metadata/MetadataMatcher.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * This file is part of the prooph/event-store.
+ * (c) 2014-2016 prooph software GmbH <contact@prooph.de>
+ * (c) 2015-2016 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Metadata;
+
+use Prooph\EventStore\Exception\InvalidArgumentException;
+
+class MetadataMatcher
```

Comment:
@oqq look at the current implementation in develop branch - this is only a container, there is no reason to have another implementation.